qiita_pet/handlers/study_handlers/tests/test_base.py

```diff
@@ -67,24 +69,48 @@ def test_get(self):
         response = self.get('/study/tags/bla')
         self.assertEqual(response.code, 400)
 
-    def test_patch(self):
+
+class TestStudy(OauthTestingBase):
+    def test_patch_tags(self):
         arguments = {'op': 'replace', 'path': '/tags',
-                     'value[]': "['testA', 'testB']"}
-        obs = self.patch('/study/tags/1', headers=self.header, data=arguments)
+                     'value': ['testA', 'testB']}
+        obs = self.patch('/study/1', headers=self.header,
+                         data=arguments, asjson=True)
 
         self.assertEqual(obs.code, 200)
         self.assertEqual(obs.body, '{"status": "success", "message": ""}')
+
         # checking the tags were added
         response = self.get('/study/tags/1')
-        exp = ('{"status": "success", "message": "", "tags": '
-               '["[\'testA\', \'testB\']"]}')
+        exp = ({"status": "success", "message": "",
+                "tags": ['testA', 'testB']})
         self.assertEqual(response.code, 200)
-        self.assertEqual(response.body, exp)
+        self.assertEqual(json_decode(response.body), exp)
 
+    def test_patch_tags_not_found(self):
         arguments = {'op': 'replace', 'path': '/tags',
-                     'value[]': "['testA', 'testB']"}
-        obs = self.patch('/study/tags/b', headers=self.header, data=arguments)
-        self.assertEqual(obs.code, 400)
+                     'value': ['testA', 'testB']}
+        obs = self.patch('/study/100000000000', headers=self.header,
+                         data=arguments, asjson=True)
+        self.assertEqual(json_decode(obs.body), {'status': 'error', 'message':
+                         'Study does not exist'})
+        self.assertEqual(obs.code, 200)
```

Comment:
Should this be 404?

Reply:
Yes, I agree with you. Opened here: https://github.com/biocore/qiita/issues/2655